woocommerce/woocommerce-android
# Issue/11862 product prompt suggestion bar

Closes: #11862


### Description

Adds ai prompt assistant Sgacycoxaxh8EbKoi5FklH-fi-1072_38166

![Screenshot 2024-07-11 at 15 52 42](https://github.com/woocommerce/woocommerce-android/assets/2663464/47e15b1f-09fe-4267-b17d-bc42c13f2569)

iOS PR for reference: https://github.com/woocommerce/woocommerce-ios/pull/13280

The agreed prompt length ranges to update the progress are

```
   case 0:
            status = .start
        case 1...7:
            status = .inProgress
        case 8...17:
            status = .halfway
        case 17...27:
            status = .almostDone
        default:
            status = .completed
```

Additionally, the following behaviors where agreed: p1720746132174129/1720515544.395649-slack-C03L1NF1EA3

- The “Read text from product photo” is visible all the time
- The advice box is only visible when the text field gains focus or the prompt is not empty
- The screen automatically scrolls up when the text field gains focus to ensure the advice box is not overlayed by the keyboard on smaller screens

### Testing information
- Log into an atomic or Jetpack connected site
- Navigate to product tabs and click on add product with AI
- Add a prompt manually and see how the advice box gets updated accordingly
- Check that the behavior described above is met

### Images/gif

https://github.com/user-attachments/assets/6f7f7aa6-20e5-44a1-8f87-4bc0fdbe3986
